packages/updating-element/src/updating-element.ts

```diff
@@ -352,173 +294,38 @@ export abstract class UpdatingElement extends HTMLElement {
-   *         },
-   *         configurable: true,
-   *         enumerable: true
-   *       }
-   *     }
-   *   }
-   *
-   * @nocollapse
-   */
-  protected static getPropertyDescriptor(
-    name: PropertyKey,
-    key: string | symbol,
-    options: PropertyDeclaration
-  ) {
-    return {
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any
-      get(): any {
-        return (this as {[key: string]: unknown})[key as string];
-      },
-      set(this: UpdatingElement, value: unknown) {
-        const oldValue = ((this as {}) as {[key: string]: unknown})[
-          name as string
-        ];
-        ((this as {}) as {[key: string]: unknown})[key as string] = value;
-        ((this as unknown) as UpdatingElement).requestUpdate(
-          name,
-          oldValue,
-          options
-        );
-      },
-      configurable: true,
-      enumerable: true,
-    };
-  }
-
-  /**
-   * Returns the property options associated with the given property.
-   * These options are defined with a PropertyDeclaration via the `properties`
-   * object or the `@property` decorator and are registered in
-   * `createProperty(...)`.
-   *
-   * Note, this method should be considered "final" and not overridden. To
-   * customize the options for a given property, override `createProperty`.
-   *
-   * @nocollapse
-   * @final
-   */
-  protected static getPropertyOptions(name: PropertyKey) {
-    return this.elementProperties!.get(name) || defaultPropertyDeclaration;
-  }
-
   /**
    * Creates property accessors for registered properties, sets up element
    * styling, and ensures any superclasses are also finalized. Returns true if
    * the element was finalized.
    * @nocollapse
    */
-  protected static finalize() {
-    if (this.hasOwnProperty(finalized)) {
-      return false;
-    }
-    this[finalized] = true;
-    // finalize any superclasses
-    const superCtor = Object.getPrototypeOf(this) as typeof UpdatingElement;
-    superCtor.finalize();
-    this.elementProperties = new Map(superCtor.elementProperties!);
-    // initialize Map populated in observedAttributes
-    this._attributeToPropertyMap = new Map();
-    // make any properties
-    // Note, only process "own" properties since this element will inherit
-    // any properties defined on the superClass, and finalization ensures
-    // the entire prototype chain is finalized.
-    if (this.hasOwnProperty(JSCompiler_renameProperty('properties', this))) {
-      const props = this.properties;
-      // support symbols in properties (IE11 does not support this)
-      const propKeys = [
-        ...Object.getOwnPropertyNames(props),
-        ...Object.getOwnPropertySymbols(props),
-      ];
-      // This for/of is ok because propKeys is an array
-      for (const p of propKeys) {
-        // note, use of `any` is due to TypeScript lack of support for symbol in
-        // index types
+  static finalize() {
+    const didFinalize = super.finalize();
```

Comment:
Nit: The meaning here is `didFinalizeThisCall` (I read with a "hasFinalized" meaning at first), so maybe `shouldFinalize` would be a clearer naming.